### PR TITLE
[OCPCLOUD-1117] Add FeatureGate check for operator

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
To prevent operator from provisioning resources in production cluster the
`ExternalCloudProvider` feature gate should be enabled in advance.

Use the following command to enable `CCCMO` and provision resources:
`oc apply -f hack/externalFeatureGate.yaml`

### CODE REVIEW

File: manifests/0000_26_cloud-controller-manager-operator_02_rbac_operator.yaml [hunk 1]
```diff
@@ -40,6 +40,7 @@ rules:
   - config.openshift.io
   resources:
   - infrastructures
+  - featureпates
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_02_rbac_operator.yaml:43]:
```suggestion
  - featuregates
```

**Review by JoelSpeed — COMMENTED** [review]
Looks great! Found a typo in the rbac that will need fixing up, but otherwise LGTM! 

Has this been manually tested on a cluster yet?

**Comment by Danil-Grigorev** [comment]
@JoelSpeed I didn't try cluster bot yet, but tested it manually in my cluster.

**Review by JoelSpeed — APPROVED** [review]
/approve

**Pull request merged**